### ISSUE TITLE
Conversion between non SI units not working

### ISSUE THREAD

**Issue by Airon90** [duckduckgo/zeroclickinfo-goodies#1767]
Source: [Reddit](https://www.reddit.com/r/duckduckgo/comments/3sup41/15_reasons_why_i_will_never_use_duckduckgo/) ([img](http://i.imgur.com/AscrDQa.png))

It seems that "[how many ounces in a cup](https://duckduckgo.com/?q=how+many+ounces+in+a+cup)" doesn't show any answer. Instead "[how many meters in a kilometer](https://duckduckgo.com/?q=how+many+meters+in+a+kilometer&ia=answer)" shows the correct answer

**Comment by mintsoft** [comment]
Duplicate of: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/1754

**Comment by Airon90** [comment]
Thank you!

**Comment by elohmrow** [comment]
@Airon90 @mintsoft @moollaza ~ #1 - thanks for your patience - I have been *very* busy and now on vacation I have time to look at all the C::P issues.  I have done a total re-write; will post back here when it is on CPAN.

I have added the alias 'cup' for 'US Cups' ... this is an interesting and slightly large problem actually ... 
[diff.txt](https://github.com/duckduckgo/zeroclickinfo-goodies/files/69667/diff.txt)
 
The problems are at least two:
"cup" isn't really defined very well (or, has multiple possible interpretations), as can be seen here: https://en.wikipedia.org/wiki/Cup_%28unit%29 ... 
some units can exist in more than one "unit_type": for example, "oz" is valid as both a "mass" and as a "volume" ... in C::P, it bails out if it finds something like this ...

So while it is now added, this needs further exploration.
